persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java

```diff
@@ -122,19 +123,28 @@ public CompletionStage<Void> start(InitializationContext ctx) {
          iceFactory = ctx.getCache().getAdvancedCache().getComponentRegistry().getComponent(InternalEntryFactory.class);
       }
 
-      ConfigurationBuilder builder = buildRemoteConfiguration(configuration, marshaller);
+      CompletionStage<RemoteCacheManager> rcmStage;
 
-      return blockingManager.supplyBlocking(() -> {
-         remoteCacheManager = new RemoteCacheManager(builder.build());
-
-         if (configuration.remoteCacheName().isEmpty())
-            remoteCache = (InternalRemoteCache<Object, Object>) remoteCacheManager.getCache();
-         else
-            remoteCache = (InternalRemoteCache<Object, Object>) remoteCacheManager.getCache(configuration.remoteCacheName());
-
-         return remoteCache.ping();
-      }, "RemoteStore-start")
-            .thenCompose(Function.identity())
+      if (isManagedRemoteCacheManager()) {
+         GlobalRemoteContainers containers = ctx.getCache().getAdvancedCache().getComponentRegistry()
+               .getGlobalComponentRegistry()
+               .getComponent(GlobalRemoteContainers.class);
+         rcmStage = containers.cacheContainer(configuration.remoteCacheContainer(), marshaller);
+      } else {
+         rcmStage = blockingManager.supplyBlocking(() -> {
+            ConfigurationBuilder builder = buildRemoteConfiguration(configuration, marshaller);
+            return new RemoteCacheManager(builder.build());
+         }, "RemoteCacheManager-create");
+      }
+      return rcmStage.thenApplyAsync(rcm -> {
```

Comment:
This should use `BlockingManager.thenApplyBlocking` so we aren't leaking out a blocking thread.